src/file-events/headers/generic_fsnotifier.h

```diff
@@ -36,13 +39,64 @@ struct FileWatcherException : public exception {
     const char* message;
 };
 
+class AbstractServer;
+
+class Command {
+public:
+    Command(){};
+    virtual ~Command(){};
+
+    void execute(AbstractServer* server) {
```

Comment:
Can we reduce the scope and not pass all of `AbstractServer` here? It seems like a better idea to me to move everything which is used on the thread to a separate object, so the command can't access/call things which it isn't suppose to call. Moreover, the comments that some methods only should be used on the thread can be removed by moving those methods to the new class - then it is only possible to call the methods from the threads.

WDYT?

Reply:
I fully agree. I wrote about this in the issue description and why I want to stop for now with the structural improvements. But what you describe is the endgame, yes.

Note that any kind of refactoring is pretty complicated, as VSCode doesn't refactor the other platforms, and in general things need to be checked on every platform.

We'll get there, but I think for now we're good.

Reply:
👍